### PR TITLE
Add list of boxes

### LINKED ISSUE #167: AVIF should have a table of boxes and box versions
Cyril has created a prototype table of required boxes/box-versions for stills and image sequences which would be a good addition to the spec. This issue will track getting that added at some point.

### PR DESCRIPTION
**Pull request by leo-barnes**
Closes #167.

This is a rebase and fixup of PR #170.
The section on box requirements for sequences has been removed until someone has taken the time to fully go through it.

***
<a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/av1-avif/pull/225.html" title="Last updated on Sep 11, 2024, 5:45 PM UTC (ea337df)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/AOMediaCodec/av1-avif/225/b9ea3d8...ea337df.html" title="Last updated on Sep 11, 2024, 5:45 PM UTC (ea337df)">Diff</a>

### CODE REVIEW

**Comment by leo-barnes** [comment]
@cconcolato
I have rebased your changes for #170 and fixed up indentation and some bikeshed warnings.

I also removed the section for sequences for now.

File: index.bs [hunk 1]
```diff
@@ -431,11 +431,361 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+        <td>&nbsp;</td>
+        <td>&nbsp;</td>
+        <td>ipma</td>
+        <td>&nbsp;</td>
+        <td>0, 1</td>
+        <td>ISOBMFF</td>
+        <td>&nbsp;</td>
+    </tr>
+    <tr>
+        <td>mdat</td>
```
Comment (y-guyon) [inline index.bs:576]: Is `mdat` required? I thought an `idat` could suffice.

File: index.bs [hunk 2]
```diff
@@ -431,11 +431,361 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+        <th>Containing Box</th>
+        <th>Description</th>
+    </tr>
+    <tr>
+        <td>dinf</td>
+        <td>&nbsp;</td>
+        <td>-</td>
+        <td>ISOBMFF</td>
+        <td>meta</td>
+        <td>&nbsp;</td>
```
Comment (y-guyon) [inline index.bs:604]: Used to indicate the location of the media information in a track

File: index.bs [hunk 3]
```diff
@@ -431,11 +431,361 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+        <td>-</td>
+        <td>HEIF</td>
+        <td>&nbsp;</td>
+        <td>Used when an image is derived from another image</td>
+    </tr>
+    <tr>
+        <td>&nbsp;</td>
+        <td>prem</td>
+        <td>-</td>
+        <td>MIAF</td>
```
Comment (y-guyon) [inline index.bs:650]: I found `prem` defined in HEIF.

File: index.bs [hunk 4]
```diff
@@ -431,11 +431,361 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+        <td>HEIF</td>
+        <td>ipco</td>
+        <td>Used to signal a mirroring applied to an image</td>
+    </tr>
+    <tr>
+        <td>clli</td>
+        <td>&nbsp;</td>
+        <td>-</td>
+        <td>ISOBMFF</td>
+        <td>ipco</td>
```
Comment (y-guyon) [inline index.bs:739]: Note that `clli` is a `Box` and not an `ItemProperty`.  
ISOBMFF says
```
aligned(8) class ItemPropertyContainerBox extends Box('ipco')
{
   Box properties[]; // boxes derived from
   // ItemProperty or ItemFullProperty, or FreeSpaceBox(es)
   // to fill the box
}
```
fortunately ISOBMFF also says
> Each item property is a Box or FullBox.

so I think this is just another inconsistency of ISOBMFF and there is no issue here.

File: index.bs [hunk 5]
```diff
@@ -431,11 +431,361 @@ A file containing a 'pict' track compliant with this profile is expected to list
...
+        <td>0</td>
+        <td>ISOBMFF</td>
+        <td>&nbsp;</td>
+    </tr>
+    <tr>
+        <td>&nbsp;</td>
+        <td>pitm</td>
+        <td>&nbsp;</td>
+        <td>&nbsp;</td>
+        <td>0, 1</td>
```
[the remark this answers is not in the document]
Comment (leo-barnes) [reply index.bs:490]: Yeah, I know. I think @podborski has opened an issue for ISOBMFF/HEIF to fix this.

**Comment by wantehchang** [comment]
@vigneshvg Vignesh: Please review. Thanks!

**Comment by vigneshvg** [comment]
yes, the list of boxes look good to me. the only thing that i found to be missing was the `ccst` box that libavif writes out for animated images: https://github.com/AOMediaCodec/libavif/blob/85979a79983573c908958cd27197290cd19ea7b6/src/write.c#L3504

**Pull request merged**